src/au3audio/internal/audioengine.cpp

```diff
@@ -47,6 +75,99 @@ static ProjectAudioIO::DefaultOptions::Scope s_defaultOptionsScope {
 void AudioEngine::init()
 {
     s_audioIOListener = std::make_shared<Au3AudioIOListener>();
+    globalContext()->currentProjectChanged().onNotify(this, [this] {
+        m_rtEffectSubscriptions.clear();
+        auto project = globalContext()->currentProject();
+        if (!project) {
+            return;
+        }
+        auto au3Project = reinterpret_cast<au::au3::Au3Project*>(project->au3ProjectPtr());
+        m_tracklistSubscription = TrackList::Get(*au3Project).Subscribe([this, w = weak_from_this()](const TrackListEvent& e) {
+            const auto lifeguard = w.lock();
+            if (lifeguard) {
+                onTrackListEvent(e);
+            }
+        });
+    });
```

Comment:
I believe there is a better place for all this. The audio engine is a low-level utility service, it is unexpected to see it accessing the global context, working with tracks, subscribing to track events...  

I think that all this added code needs to be moved to the `effects_base` module, adding a new interface, something like `IRealtimeEffects` and its implementation.